mobile/src/context/BookContext.tsx

```diff
@@ -1,191 +1,57 @@
-        createdAt: '3000',
-      },
-      {
-        id: 'e',
-        title: 'bar',
-        image: 'https://placekitten.com/200/304',
-        createdAt: '1000',
-      },
-      {
-        id: 'f',
-        title: 'bar',
-        image: 'https://placekitten.com/200/305',
-        createdAt: '1000',
-      },
-      {
-        id: 'g',
-        title: 'bar',
-        image: 'https://placekitten.com/200/306',
-        createdAt: '1000',
-      },
-      {
-        id: 'h',
-        title: 'bar',
-        image: 'https://placekitten.com/200/307',
-        createdAt: '1000',
-      },
-      {
-        id: 'i',
-        title: 'bar',
-        image: 'https://placekitten.com/200/310',
-        createdAt: '1000',
-      },
-      {
-        id: 'j',
-        title: 'foo',
-        image: 'https://placekitten.com/200/300',
-        createdAt: '10000',
-      },
-      {
-        id: 'k',
-        title: 'bar',
-        image: 'https://placekitten.com/200/301',
-        createdAt: '1500',
-      },
-      {
-        id: 'l',
-        title: 'bar',
-        image: 'https://placekitten.com/200/302',
-        createdAt: '2000',
-      },
-      {
-        id: 'm',
-        title: 'bar',
-        image: 'https://placekitten.com/200/303',
-        createdAt: '3000',
-      },
-      {
-        id: 'n',
-        title: 'bar',
-        image: 'https://placekitten.com/200/304',
-        createdAt: '1000',
-      },
-      {
-        id: 'o',
-        title: 'bar',
-        image: 'https://placekitten.com/200/305',
-        createdAt: '1000',
-      },
-      {
-        id: 'p',
-        title: 'bar',
-        image: 'https://placekitten.com/200/306',
-        createdAt: '1000',
-      },
-      {
-        id: 'q',
-        title: 'bar',
-        image: 'https://placekitten.com/200/307',
-        createdAt: '1000',
-      },
-      {
-        id: 'r',
-        title: 'bar',
-        image: 'https://placekitten.com/200/310',
-        createdAt: '1000',
-      },
-      {
-        id: 's',
-        title: 'foo',
-        image: 'https://placekitten.com/200/300',
-        createdAt: '10000',
-      },
-      {
-        id: 't',
-        title: 'bar',
-        image: 'https://placekitten.com/200/301',
-        createdAt: '1500',
-      },
-      {
-        id: 'u',
-        title: 'bar',
-        image: 'https://placekitten.com/200/302',
-        createdAt: '2000',
-      },
-      {
-        id: 'v',
-        title: 'bar',
-        image: 'https://placekitten.com/200/303',
-        createdAt: '3000',
-      },
-      {
-        id: 'w',
-        title: 'bar',
-        image: 'https://placekitten.com/200/304',
-        createdAt: '1000',
-      },
-      {
-        id: 'x',
-        title: 'bar',
-        image: 'https://placekitten.com/200/305',
-        createdAt: '1000',
-      },
-
-    ];
-    dispatch({ type: 'BOOKS_LOADED', payload: books });
+    client.getBooks().then((res) => {
+      dispatch({ type: 'BOOKS_LOADED', payload: res, loading: false });
```

Comment:
Typically dispatch only takes a `type` and a `payload`. `payload` should be reserved for extra data, like API responses. Instead, you should set the `loading` bool within the reducer.